scripts/classificationloss.py

```diff
@@ -0,0 +1,220 @@
+import sys, os
+
+from PIL import Image
+from podm import metrics
+import numpy as np
+
+# Accroding to: https://pyimagesearch.com/2022/05/02/mean-average-precision-map-using-the-coco-evaluator/
+# This ia also helpful: https://github.com/yfpeng/object_detection_metrics/tree/master/src/podm
+
+
+def IOU(boxes1, boxes2):
+    x11, y11, x12, y12 = np.split(boxes1, 4, axis=1)
+    x21, y21, x22, y22 = np.split(boxes2, 4, axis=1)
+    xA = np.maximum(x11, np.transpose(x21))
+    yA = np.maximum(y11, np.transpose(y21))
+    xB = np.minimum(x12, np.transpose(x22))
+    yB = np.minimum(y12, np.transpose(y22))
+    interArea = np.maximum((xB - xA), 0) * np.maximum((yB - yA), 0)
+    boxAArea = (x12 - x11) * (y12 - y11)
+    boxBArea = (x22 - x21) * (y22 - y21)
+    iou = interArea / (boxAArea + np.transpose(boxBArea) - interArea)
+    return iou
+
+
+def boxes_xywh_to_xyxy(boxes):
+    res = np.copy(boxes[:, :])
+    res[:, 2] += res[:, 0]
+    res[:, 3] += res[:, 1]
+    return res
+
+
+def box_xywh_to_xyxy(boxes):
+    res = np.copy(boxes)
+    res[2] += res[0]
+    res[3] += res[1]
+    return res
```

Comment:
If I'm understanding this right, you import in the yolo5 format from roboflow, convert to some custom format, then convert again with this to allow you to use the existing podm tools. Why not just use COCO and never have to convert anything?